packages/apollo-cli/src/git.ts

```diff
@@ -0,0 +1,55 @@
+import * as path from "path";
+import * as fs from "fs";
+import * as ci from "env-ci";
+import { gitToJs } from "git-parse";
+import * as git from "git-rev-sync";
+import { pickBy, identity } from "lodash";
+
+const findGitRoot = (start?: string | string[]): string => {
+  start = start || module.parent!.filename;
+  if (typeof start === "string") {
+    if (start[start.length - 1] !== path.sep) start += path.sep;
+    start = start.split(path.sep);
+  }
+  if (!start.length) throw new Error(".git/ not found in path");
```

Comment:
lets pull in a commit (or manually make the change) from upstream apollo-cli here. I make a fix that didn't throw to make git not required

https://github.com/apollographql/apollo-cli/commit/28c7d298adc0d48292c8f259b1670f00d6e61050